src/content/documentation/develop/extensions-and-the-add-on-id.md

```diff
@@ -56,25 +55,31 @@ This article describes how add-on IDs are handled for extensions built with WebE
 
 ## When do I need an add-on ID?
 
-All [Manifest V3 extensions](/documentation/develop/manifest-v3-migration-guide/) need an add-on ID in their manifest.json when submitted to AMO.
+### Signing and publishing
+
+All signed add-ons have a unique ID. 
+
+For [Manifest V3 extensions](/documentation/develop/manifest-v3-migration-guide/) you must create the add-on ID and add it to your extension's manifest.json file before it's submitted to AMO ([online](/documentation/publish/submitting-an-add-on/) or using [`web-ext sign`](/documentation/develop/web-ext-command-reference/#web-ext-sign) or the [AMO API create endpoint](https://mozilla.github.io/addons-server/topics/api/addons.html#create)). It's recommended that you set your extension ID when you begin development.
+
+You can define an ID for a Manifest V2 extension in its manifest.json file before submission. If you don't, AMO assigns your extension an ID when you submit it to AMO (online or using `web-ext sign` or the AMO API create endpoint).
```

Comment:
Let's recommend developers to always add their own ID explicitly in the manifest instead of making it a MV2 vs. MV3 thing.

Reply:
@willdurand let me know what you think of the revision.